src/ystdlib/io_interface/ReaderInterface.hpp

```diff
@@ -10,140 +13,85 @@
 namespace ystdlib::io_interface {
 class ReaderInterface {
 public:
-    // Types
-    class OperationFailed : public std::exception {
-    public:
-        OperationFailed(ErrorCode error_code) {}
-    };
+    // Constructor
+    ReaderInterface() = default;
 
-    // Destructor
-    virtual ~ReaderInterface() = default;
+    // Delete copy constructor and assignment operator
+    ReaderInterface(ReaderInterface const&) = delete;
+    auto operator=(ReaderInterface const&) -> ReaderInterface& = delete;
 
-    // Methods
-    virtual ErrorCode try_read(char* buf, size_t num_bytes_to_read, size_t& num_bytes_read) = 0;
-    virtual ErrorCode try_seek_from_begin(size_t pos) = 0;
-    virtual ErrorCode try_get_pos(size_t& pos) = 0;
+    // Default move constructor and assignment operator
+    ReaderInterface(ReaderInterface&&) noexcept = default;
+    auto operator=(ReaderInterface&&) noexcept -> ReaderInterface& = default;
 
-    /**
-     * Tries to read up to the next delimiter and stores it in the given string.
-     * NOTE: Implementations should override this if they can achieve better performance.
-     * @param delim The delimiter to stop at
-     * @param keep_delimiter Whether to include the delimiter in the output string or not
-     * @param append Whether to append to the given string or replace its contents
-     * @param str The string read
-     * @return ErrorCode_Success on success
-     * @return Same as ReaderInterface::try_read otherwise
-     */
-    virtual ErrorCode
-    try_read_to_delimiter(char delim, bool keep_delimiter, bool append, std::string& str);
+    // Destructor
+    virtual ~ReaderInterface() = default;
 
-    /**
-     * Reads up to a given number of bytes
+    // Methods implementing ReaderInterface
```

Comment:
I think saying "implement" here is confusing considering these are all virtual and definitions.
```suggestion
    // Methods
```